code/game/objects/structures/bonfire.dm

```diff
@@ -122,6 +122,17 @@
 	if(burning & !grill)
 		bonfire_burn()
 
+	//Not currently burning, let's see if we can ignite it.
+	else if(isliving(entered))
```

Comment:
Since you're here, just noticed the above conditional uses the incorrect `and` (bitwise and) which could cause unpredictable behavior, mind cleaning the rest of this up?

(Also, if burning is `true` and grill is `true`, the visible messages below will spam every time a person enters the bonfire, even if it's already on fire)